internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -835,233 +841,380 @@ func resourceCosmosDbAccountCreate(d *pluginsdk.ResourceData, meta interface{})
+			if expandedIdentity.Type != documentdb.ResourceIdentityTypeNone {
+				log.Printf("[INFO] Updating AzureRM Cosmos DB Account: Updating 'Identity' to %q", expandedIdentity.Type)
+
+				identity := documentdb.DatabaseAccountUpdateParameters{
+					Identity: expandedIdentity,
+				}
+
+				// Update the database...
+				err = resourceCosmosDbAccountApiUpdate(client, ctx, id.ResourceGroup, id.Name, identity, d)
+				if err != nil {
+					return fmt.Errorf("updating 'identity' %q: %+v", id, err)
 				}
-				delete(oldLocationsMap, *l.LocationName)
-				removedOne = true
-				continue
 			}
-		}
-	}
+		} else {
+			log.Printf("[INFO] [SKIP] AzureRM Cosmos DB Account: Updating 'Identity' [NO CHANGE]")
+		}
+
+		// NOTE: updateDefaultIdentity now has a default value of 'FirstPartyIdentity'... This value now gets
+		//       triggered if the default value does not match the value in Azure...
+		//
+		// NOTE: When you change the 'Identity', the 'DefaultIdentity' will be set to 'undefined', so if you change
+		//       the identity you must also update the 'DefaultIdentity' as well...
+		if updateDefaultIdentity || identityChanged {
+			// This will now return the default of 'FirstPartyIdentity' if it
+			// is not set in the config, which is correct.
+			configDefaultIdentity := d.Get("default_identity_type").(string)
+			if identityChanged {
+				log.Printf("[INFO] Updating AzureRM Cosmos DB Account: Updating 'DefaultIdentity' to %q because the 'Identity' was changed to %q", configDefaultIdentity, expandedIdentity.Type)
+			} else {
+				log.Printf("[INFO] Updating AzureRM Cosmos DB Account: Updating 'DefaultIdentity' to %q", configDefaultIdentity)
+			}
 
-	if removedOne {
-		locationsUnchanged := make([]documentdb.Location, 0, len(oldLocationsMap))
-		for _, value := range oldLocationsMap {
-			locationsUnchanged = append(locationsUnchanged, value)
-		}
+			// PATCH instead of PUT...
+			defaultIdentity := documentdb.DatabaseAccountUpdateParameters{
+				DatabaseAccountUpdateProperties: &documentdb.DatabaseAccountUpdateProperties{
+					DefaultIdentity: pointer.To(configDefaultIdentity),
+				},
+			}
 
-		account.DatabaseAccountCreateUpdateProperties.Locations = &locationsUnchanged
-		if err = resourceCosmosDbAccountApiUpsert(client, ctx, id.ResourceGroup, id.Name, account, d); err != nil {
-			return fmt.Errorf("removing %s renamed locations: %+v", id, err)
+			// Update the database...
+			err = resourceCosmosDbAccountApiUpdate(client, ctx, id.ResourceGroup, id.Name, defaultIdentity, d)
+			if err != nil {
+				return fmt.Errorf("updating 'default_identity_type' %q: %+v", id, err)
+			}
 		}
-	}
 
-	// add any new/renamed locations
-	account.DatabaseAccountCreateUpdateProperties.Locations = &newLocations
-	err = resourceCosmosDbAccountApiUpsert(client, ctx, id.ResourceGroup, id.Name, account, d)
-	if err != nil {
-		return fmt.Errorf("updating %s locations: %+v", id, err)
+		d.SetId(id.ID())
```

Comment:
whilst outside the scope of this PR - the ID shouldn't change during the Update, so we can remove this:

```suggestion
```

Reply:
It was like that, so I didn't know if it was some special logic for the CosmosDB or not so I didn't touch it. 🙂 

Fixed.